server/container/guice/protocols/imap/src/main/java/org/apache/james/modules/protocols/ImapPackage.java

```diff
@@ -0,0 +1,314 @@
+import org.apache.james.imap.processor.SetMetadataProcessor;
+import org.apache.james.imap.processor.SetQuotaProcessor;
+import org.apache.james.imap.processor.StartTLSProcessor;
+import org.apache.james.imap.processor.StatusProcessor;
+import org.apache.james.imap.processor.StoreProcessor;
+import org.apache.james.imap.processor.SubscribeProcessor;
+import org.apache.james.imap.processor.UnselectProcessor;
+import org.apache.james.imap.processor.UnsubscribeProcessor;
+import org.apache.james.imap.processor.XListProcessor;
+import org.apache.james.imap.processor.base.ImapResponseMessageProcessor;
+import org.apache.james.imap.processor.fetch.FetchProcessor;
+import org.apache.james.utils.ClassName;
+
+import com.google.common.collect.ImmutableList;
+
+public interface ImapPackage {
+    Collection<ClassName> processors();
+
+    Collection<ClassName> decoders();
+
+    Collection<ClassName> encoders();
+
+    class Impl implements ImapPackage {
+        private final ImmutableList<ClassName> processors;
+        private final ImmutableList<ClassName> decoders;
+        private final ImmutableList<ClassName> encoders;
+
+        public Impl(ImmutableList<ClassName> processors,
+                    ImmutableList<ClassName> decoders,
+                    ImmutableList<ClassName> encoders) {
+            this.processors = processors;
+            this.decoders = decoders;
+            this.encoders = encoders;
+        }
+
+        @Override
+        public Collection<ClassName> processors() {
+            return processors;
+        }
+
+        @Override
+        public Collection<ClassName> decoders() {
+            return decoders;
+        }
+
+        @Override
+        public Collection<ClassName> encoders() {
+            return encoders;
+        }
+    }
+
+    static ImapPackage and(Collection<ImapPackage> packages) {
+        return new ImapPackage() {
+            @Override
+            public Collection<ClassName> processors() {
+                return packages.stream()
+                    .flatMap(p -> p.processors().stream())
+                    .collect(ImmutableList.toImmutableList());
+            }
+
+            @Override
+            public Collection<ClassName> decoders() {
+                return packages.stream()
+                    .flatMap(p -> p.decoders().stream())
+                    .collect(ImmutableList.toImmutableList());
+            }
+
+            @Override
+            public Collection<ClassName> encoders() {
+                return packages.stream()
+                    .flatMap(p -> p.encoders().stream())
+                    .collect(ImmutableList.toImmutableList());
+            }
+        };
+    }
+
+    ImapPackage DEFAULT = and(ImmutableList.of(new DefaultNoAuth(), new DefaultAuth()));
+
+    class DefaultNoAuth extends Impl {
+        public DefaultNoAuth() {
+            super(
+                ImmutableList.of(
+                    CapabilityProcessor.class,
```

Comment:
If we are rebased on master we need to add here IdProcessor + its encoder and command parser

Reply:
`org.apache.james.imap.processor.IdProcessor` already
yes, i forgot encoder/decoder